adagrams/game.py

```diff
@@ -1,13 +1,158 @@
 from random import randint
 
+HAND_SIZE = 10
+
+BONUS_MIN_LENGTH = 7
+
+LENGTH_BONUS_POINTS = 8
+
+PAIR_WORD = 0
+PAIR_SCORE = 1
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2,
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+
+  all_letter_occurencies = []
+
+  hand_of_letters = []
+
+  for letter, frequency in LETTER_POOL.items():
+    all_letter_occurencies.extend([letter] *frequency)
+
+  while len(hand_of_letters) < HAND_SIZE:
+    # random_index = randint(0, len(all_letter_occurencies)-1)
+    # random_letter = all_letter_occurencies[random_index]
+    # hand_of_letters.append(random_letter)
+    # all_letter_occurencies.pop(random_index)
+
+    index = randint(0, len(all_letter_occurencies)-1)
+    all_letter_occurencies[index], all_letter_occurencies[-1] = all_letter_occurencies[-1], all_letter_occurencies[index]
+    popped = all_letter_occurencies.pop()
+    hand_of_letters.append(popped)
```

Comment:
👍 Nice switch over to the swap trick for popping a value without incurring a linear cost. For the size of our data, we're not too worried about the cost of the previous logic, but it's nice to pravtice this approach when we can. We just need to be certain that the order of the values in the list truly doesn't matter.